pact/coin-contract/v4/coin-v4.pact

```diff
@@ -497,40 +529,41 @@
 
     (step
       (with-capability (DEBIT sender)
+        (with-capability
+          (TRANSFER-XCHAIN sender receiver amount target-chain)
 
-        (validate-account sender)
-        (validate-account receiver)
+          (validate-account sender)
+          (validate-account receiver)
 
-        (enforce (!= "" target-chain) "empty target-chain")
-        (enforce (!= (at 'chain-id (chain-data)) target-chain)
-          "cannot run cross-chain transfers to the same chain")
+          (enforce (!= "" target-chain) "empty target-chain")
+          (enforce (!= (at 'chain-id (chain-data)) target-chain)
+            "cannot run cross-chain transfers to the same chain")
 
-        (enforce (> amount 0.0)
-          "transfer quantity must be positive")
+          (enforce (> amount 0.0)
+            "transfer quantity must be positive")
 
-        (enforce-unit amount)
+          (enforce-unit amount)
 
-        ;; step 1 - debit delete-account on current chain
-        (debit sender amount)
+          ;; step 1 - debit delete-account on current chain
+          (debit sender amount)
+          (emit-event (TRANSFER-XCHAIN sender "" amount target-chain))
 
-        (emit-event (TRANSFER sender "" amount))
-
-        (let
-          ((crosschain-details:object{crosschain-schema}
-            { "receiver" : receiver
-            , "receiver-guard" : receiver-guard
-            , "amount" : amount
-            }))
-          (yield crosschain-details target-chain)
-          )))
+          (let
+            ((crosschain-details:object{crosschain-schema}
+              { "receiver" : receiver
+              , "receiver-guard" : receiver-guard
+              , "amount" : amount
+              }))
+            (yield crosschain-details target-chain)
+            ))))
 
     (step
       (resume
         { "receiver" := receiver
         , "receiver-guard" := receiver-guard
         , "amount" := amount
         }
-        (emit-event (TRANSFER "" receiver amount))
+        (emit-event (TRANSFER-XCHAIN "" receiver amount target-chain))
```

Comment:
Keep the `TRANSFER` create event. 

Instead of emitting `TRANSFER-XCHAIN`, we should probably emit a `TRANSFER-XCHAIN-RECD` event in step 2, with the same signature but with `source-chain` instead of `target-chain`. 

Note you'll have to `yield` the source chain value to recover in the second step. This requires a new field in  `crosschain-schema` but that should be fine to upgrade because we don't persist it.